api/api-pipeline-operations-v1/src/test/groovy/com/thoughtworks/go/apiv1/pipelineoperations/representers/PipelineScheduleOptionsRepresenterTest.groovy

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.pipelineoperations.representers
+
+import com.thoughtworks.go.api.util.GsonTransformer
+import org.junit.Test
+
+import static groovy.test.GroovyAssert.shouldFail
+import static org.assertj.core.api.Assertions.assertThat
+
+class PipelineScheduleOptionsRepresenterTest {
+  @Test
+  void 'should deserialize'() {
+    def scheduleOptionsJson = new HashMap<>()
+    def material1 = new HashMap()
+    material1.put("fingerprint", "fingerprint1")
+    material1.put("revision", "revision1")
+    def material2 = new HashMap()
+    material2.put("fingerprint", "fingerprint2")
+    material2.put("revision", "revision2")
+    def plainVar1 = new HashMap<>()
+    plainVar1.put("name", "VAR1")
+    plainVar1.put("value", "val1")
+    def plainVar2 = new HashMap<>()
+    plainVar2.put("name", "VAR2")
+    plainVar2.put("value", "val2")
+    def secureVar1 = new HashMap<>()
+    secureVar1.put("name", "SEC_VAR1")
+    secureVar1.put("value", "secval1")
+    secureVar1.put("secure", true)
+    def secureVar2 = new HashMap<>()
+    secureVar2.put("name", "SEC_VAR2")
+    secureVar2.put("encrypted_value", "encrypted_secval2")
+    secureVar2.put("secure", true)
+
+    scheduleOptionsJson.put("update_materials_before_scheduling", true)
+    scheduleOptionsJson.put("materials", Arrays.asList(material1, material2))
+    scheduleOptionsJson.put("environment_variables", Arrays.asList(plainVar1, plainVar2, secureVar1, secureVar2))
+
+
+    def jsonReader = GsonTransformer.instance.jsonReaderFrom(scheduleOptionsJson)
+    def pipelineScheduleOptions = PipelineScheduleOptionsRepresenter.fromJSON(jsonReader)
+
+    assertThat(pipelineScheduleOptions.shouldPerformMDUBeforeScheduling(), is(true))
+
+    assertThat(pipelineScheduleOptions.getPlainTextEnvironmentVariables().size(), is(2))
```

Comment:
assertThat(foo).hasSize(2). The 2nd matcher argument is really only needed if there's no assertion method.

It's probably just simpler (and possibly clearer) to just assert on the entire hash, instead of piecemeal things in the json. See `EncryptedValueRepresenterTest` for an example

Reply:
> assertThat(foo).hasSize(2)
Thanks for the tip 

This is a deserialization test, so I need to assert on the deserialized java object not the json